### ISSUE TITLE
Support (KMS) SSE

### ISSUE THREAD

**Issue by copumpkin** [dvassallo/s3-benchmark#2]
It would be fascinating to see how S3's SSE (both KMS and the old-fashioned flavor) affect throughput and latencies. Might require some finagling to figure out how/if per-object data keys are getting cached somewhere in S3 after getting KMS to decrypt the data key.

**Comment by dvassallo** [comment]
Yes, good idea. Should be easy to do. S3 will likely cache the keys, but it would still be interesting to see if the download rate gets affected beyond any hit to the first byte latency. I suspect the download rate won't change though. S3's sends you data at about 93 MB/s per thread, and decryption shouldn't be anywhere close to the bottleneck at that rate.

**Comment by copumpkin** [comment]
Yeah, I'd hope not. It would also be interesting to see what kind of latency (distribution) it adds to retrieval as well though. Anyway, thanks for the neat tool and all the data!
